quinn/src/platform/unix.rs

```diff
@@ -334,7 +338,11 @@ fn recv(
 
 /// Returns the platforms UDP socket capabilities
 pub fn caps() -> UdpCapabilities {
-    *CAPABILITIES
+    let mut caps = *CAPABILITIES;
+    if SAW_IO_ERROR.load(Ordering::Relaxed) {
```

Comment:
Should capabilities be updated instead of doing this? The fallback generating an addditional atomic operation is a bit inconvenient. I'm wondering what the cheapest way to get this done is. `CAPABILITIES` being a `static mut`? It doesn't really matter if the access is too slowly propagated as long as the new value gets visible at some time.